awx/ui_next/src/screens/Setting/LDAP/LDAPEdit/LDAPEdit.jsx

```diff
@@ -1,25 +1,218 @@
+import {
+  BooleanField,
+  ChoiceField,
+  EncryptedField,
+  InputField,
+  ObjectField,
+} from '../../shared/SharedFields';
+import { formatJson } from '../../shared/settingUtils';
+import useModal from '../../../../util/useModal';
+import useRequest from '../../../../util/useRequest';
+import { SettingsAPI } from '../../../../api';
+
+function LDAPEdit() {
+  const history = useHistory();
+  const { isModalOpen, toggleModal, closeModal } = useModal();
+  const { PUT: options } = useSettings();
+  const {
+    params: { category },
+  } = useRouteMatch('/settings/ldap/:category/edit');
+  const ldapCategory =
+    category === 'default' ? 'AUTH_LDAP_' : `AUTH_LDAP_${category}_`;
+
+  const { isLoading, error, request: fetchLDAP, result: ldap } = useRequest(
+    useCallback(async () => {
+      const { data } = await SettingsAPI.readCategory('ldap');
+
+      const mergedData = {};
+      Object.keys(data).forEach(key => {
+        if (!options[key]) {
+          return;
+        }
+        mergedData[key] = options[key];
+        mergedData[key].value = data[key];
+      });
+
+      return mergedData;
+    }, [options]),
+    null
+  );
+
+  useEffect(() => {
+    fetchLDAP();
+  }, [fetchLDAP]);
+
+  const { error: submitError, request: submitForm } = useRequest(
+    useCallback(
+      async values => {
+        await SettingsAPI.updateAll(values);
+        history.push(`/settings/ldap/${category}/details`);
+      },
+      [history, category]
+    ),
+    null
+  );
+
+  const handleSubmit = async form => {
+    await submitForm({
+      [`${ldapCategory}BIND_DN`]: form[`${ldapCategory}BIND_DN`],
+      [`${ldapCategory}BIND_PASSWORD`]: form[`${ldapCategory}BIND_PASSWORD`],
+      [`${ldapCategory}DENY_GROUP`]: form[`${ldapCategory}DENY_GROUP`],
+      [`${ldapCategory}GROUP_TYPE`]: form[`${ldapCategory}GROUP_TYPE`],
+      [`${ldapCategory}REQUIRE_GROUP`]: form[`${ldapCategory}REQUIRE_GROUP`],
+      [`${ldapCategory}SERVER_URI`]: form[`${ldapCategory}SERVER_URI`],
+      [`${ldapCategory}START_TLS`]: form[`${ldapCategory}START_TLS`],
+      [`${ldapCategory}USER_DN_TEMPLATE`]: form[
+        `${ldapCategory}USER_DN_TEMPLATE`
+      ],
+      [`${ldapCategory}GROUP_SEARCH`]: formatJson(
+        form[`${ldapCategory}GROUP_SEARCH`]
+      ),
+      [`${ldapCategory}GROUP_TYPE_PARAMS`]: formatJson(
+        form[`${ldapCategory}GROUP_TYPE_PARAMS`]
+      ),
+      [`${ldapCategory}ORGANIZATION_MAP`]: formatJson(
+        form[`${ldapCategory}ORGANIZATION_MAP`]
+      ),
+      [`${ldapCategory}TEAM_MAP`]: formatJson(form[`${ldapCategory}TEAM_MAP`]),
+      [`${ldapCategory}USER_ATTR_MAP`]: formatJson(
+        form[`${ldapCategory}USER_ATTR_MAP`]
+      ),
+      [`${ldapCategory}USER_FLAGS_BY_GROUP`]: formatJson(
+        form[`${ldapCategory}USER_FLAGS_BY_GROUP`]
+      ),
+      [`${ldapCategory}USER_SEARCH`]: formatJson(
+        form[`${ldapCategory}USER_SEARCH`]
+      ),
+    });
+  };
+
+  const handleRevertAll = async () => {
```

Comment:
@marshmalien Is there a way to incorporate the `ldapCategory` into the revert handler? The current implementation will revert the settings across the different ldap configs (default->5). I think that might be unexpected for some, or at the very least inconsistent as compared to the behavior of the other forms. The [github auth settings](https://github.com/ansible/awx/pull/8761/files), for example, reverts only what's on the tab.

Reply:
Yes, good catch and an oversight on my part. I agree, this would be unexpected and we should definitely revert them by category.

Reply:
Fixed in 79c0529